packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -481,7 +485,8 @@ export class GraphQLTransform {
         functions[templateName.replace('functions/', '')] = template;
       }
     }
-    const schema = fileAssets.get('schema.graphql') || '';
+    const compiledSchema = fileAssets.get('schema.graphql') || '';
+    const schema = removeAmplifyInputDefinition(compiledSchema);
```

Comment:
It does strip it away but not from the input schema but from the compiled schema. Post compilation if we have the `Amplify` input object in the schema it will fail during deployment since that type is not recognized by AppSync.